03portfolio-assignment/scripts/blogView.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// create object for all of the functions that control what blog articles you see
+var blogview = {};
+
+// Populate filters with the authors and categories from the blogsource
+blogview.populateFilters = function() {
+  $('article').not('.template').each(function() {
+    var authorName, category, optionTag;
+    authorName = $(this).find('address a').text();
+    optionTag = '<option value="' + authorName + '">' + authorName + '</option>';
+    $('#author-filter').append(optionTag);
+    category = $(this).attr('data-category');
+    console.log("cat"+ $(this).attr('data-category'));
+    optionTag = '<option value="' + category + '">' + category + '</option>';
+    if ($('#category-filter option[value="' + category + '"]').length === 0) {
+      $('#category-filter').append(optionTag);
+    }
+  });
+};
+
+// changes the articles shown by selection of author
+blogview.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    var selection = $(this).val();
+    if (selection) {
+      $('article').hide();
+      $('article[data-author="' + selection + '"]').fadeIn(800);
+    } else {
+      $('article').not('.template').show();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+// changes the articles shown by selection of category
+blogview.handleCategoryFilter = function() {
+  $('#category-filter').on('change', function () {
+    var selection = $(this).val();
+    if (selection) {
+      $('article').hide();
+      $('article[data-category="' + selection + '"]').fadeIn(800);
+    } else {
+      $('article').not('.template').show();
+    }
+    $('#author-filter').val('');
+  })
+};
+
+// changes the class so you only see the table content you choose base on main nav
+blogview.handleMainNav = function () {
+  $('.main-nav').on('click', '.tab', function() {
+    $('.tab-content').hide();
+    var selection = $(this).attr('data-content');
+    if (selection === 'articles') {
```

Comment:
This If/Else If/Else If statement block can be slimmed down. Also, note that the final statement should be an Else, not an Else If...